### PR TITLE
KRV-4943 storage delete and get

### PR DESCRIPTION
**Pull request by EvgenyUglov**
# Description
This PR adds functionality for `karavictl storage delete` and `karavictl storage get` in the helm deployment.

## Storage Get

1. karavictl builts storage service client and makes GET request
2. storage-service queries the `karavi-storage-secret` secret to get configured storage
3. storage-service retrieves the storage system using storage type and system-id provided
4. storage-service marshals the storage data and returns a []byte response
5. karavictl receives the response and formats a json output

## Storage Delete

1. karavictl builts storage service client and makes DELETE request
2. storage-service queries the `karavi-storage-secret` secret to get configured storage
3. storage-service finds and delete the storage data using storage type and system-id provided and applies the new data to the `karavi-storage-secret`

# GitHub Issues
List the GitHub issues impacted by this PR:

| GitHub Issue # |
| -------------- |
|https://github.com/dell/csm/issues/261|

# Checklist:

- [x] I have performed a self-review of my own code to ensure there are no formatting, vetting, linting, or security issues
- [x] I have verified that new and existing unit tests pass locally with my changes
- [x] I have not allowed coverage numbers to degenerate
- [ ] I have maintained at least 90% code coverage
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Backward compatibility is not broken

# How Has This Been Tested?

This has been tested with unit tests and manual testing in the helm deployment.

```
?       karavi-authorization/cmd/karavictl      [no test files]
ok      karavi-authorization/cmd/karavictl/cmd  0.592s  coverage: 65.7% of statements
ok      karavi-authorization/cmd/proxy-server   0.026s  coverage: 4.0% of statements
?       karavi-authorization/cmd/role-service   [no test files]
?       karavi-authorization/cmd/sidecar-proxy  [no test files]
?       karavi-authorization/cmd/storage-service        [no test files]
ok      karavi-authorization/cmd/tenant-service 0.014s  coverage: 8.7% of statements
ok      karavi-authorization/deploy     0.055s  coverage: 87.5% of statements
?       karavi-authorization/internal/decision  [no test files]
ok      karavi-authorization/internal/k8s       0.021s  coverage: 51.0% of statements
ok      karavi-authorization/internal/powerflex 9.074s  coverage: 92.6% of statements
ok      karavi-authorization/internal/proxy     7.700s  coverage: 66.1% of statements
ok      karavi-authorization/internal/quota     0.099s  coverage: 90.1% of statements
ok      karavi-authorization/internal/role-service      0.014s  coverage: 79.2% of statements
ok      karavi-authorization/internal/role-service/roles        0.010s  coverage: 87.4% of statements
ok      karavi-authorization/internal/role-service/validate     0.053s  coverage: 79.4% of statements
ok      karavi-authorization/internal/storage-service   0.009s  coverage: 92.7% of statements
ok      karavi-authorization/internal/storage-service/validate  0.089s  coverage: 91.2% of statements
ok      karavi-authorization/internal/tenantsvc 1.455s  coverage: 82.3% of statements
ok      karavi-authorization/internal/token     0.006s  coverage: 100.0% of statements
ok      karavi-authorization/internal/token/jwx 0.008s  coverage: 71.7% of statements
?       karavi-authorization/internal/types     [no test files]
ok      karavi-authorization/internal/web       0.011s  coverage: 6.2% of statements
?       karavi-authorization/pb [no test files]
```

### CODE REVIEW

File: internal/storage-service/service.go [hunk 1]
```diff
@@ -206,6 +206,77 @@ func (s *Service) Update(ctx context.Context, req *pb.StorageUpdateRequest) (*pb
...
+	if !ok {
+		return nil, fmt.Errorf("error: storage of type %s is missing", req.StorageType)
+	}
+
+	s.log.Debug("Check the requested system ID exists")
+	if _, systemIDExists := systemType[req.SystemId]; !systemIDExists {
+		return nil, fmt.Errorf("error: system with ID %s does not exist", req.SystemId)
+	}
+
+	delete(systemType, req.SystemId)
```
Comment (atye) [inline internal/storage-service/service.go:234]: Could you add a log statement above this line regarding updating the storage data? Something like this:

https://github.com/dell/karavi-authorization/blob/1c6ec905cfcbc5c035df599d91716fecf24c8ee8/internal/storage-service/service.go#L199

File: internal/storage-service/service.go [hunk 2]
```diff
@@ -206,6 +206,77 @@ func (s *Service) Update(ctx context.Context, req *pb.StorageUpdateRequest) (*pb
...
+	}
+
+	s.log.Debug("Check the requested system ID exists")
+	if _, systemIDExists := systemType[req.SystemId]; !systemIDExists {
+		return nil, fmt.Errorf("error: system with ID %s does not exist", req.SystemId)
+	}
+
+	s.log.Debug("JSON marshaling configured storage")
+	system := systemType[req.SystemId]
+	system.Password = "(omitted)"
```
Comment (atye) [inline internal/storage-service/service.go:271]: The password is scrubbed on the client (karavictl) side. I'd suggest to remove this line or the one in karavictl.

https://github.com/dell/karavi-authorization/blob/1c6ec905cfcbc5c035df599d91716fecf24c8ee8/cmd/karavictl/cmd/storage_get.go#L104

Comment (atye) [reply internal/storage-service/service.go:271]: Per discussion, the karavictl code is necessary for the rpm functionality.

**Review by atye — APPROVED** [review]

**Review by hoppea2 — APPROVED** [review]

**Comment by EvgenyUglov** [comment]
run e2e test

**Comment by EvgenyUglov** [comment]
run e2e test

**Comment by EvgenyUglov** [comment]
run e2e test

**Comment by EvgenyUglov** [comment]
run e2e test

**Pull request merged**